src/mqt/predictor/ml/Predictor.py

```diff
@@ -126,8 +128,8 @@ def compile_all_circuits_devicewise(
         logger.info("Processing: " + device_name + " for " + figure_of_merit)
         rl_pred = rl.Predictor(figure_of_merit=figure_of_merit, device_name=device_name)
 
-        dev_index = rl.helper.get_device_index_of_device(device_name)
-        dev_max_qubits = rl.helper.get_devices()[dev_index]["max_qubits"]
+        dev_index = next((i for i, dev in enumerate(self.devices) if dev.name == device_name), None)
```

Comment:
Just as an idea to be discussed: Would it make sense to have the `get_available_devices` return a `dict[dev_name:str, Device` instead such that we can get rid of all the occurrences where the `dev_index` is determined manually? 

When we would need the device at a given index, we could call `list(<dev_dict>)[index]`. Let me know what you think about it, @flowerthrower .

Reply:
Thanks for the tip. There is already a rather inexpensive `get_available_device_names()` function available in MQTBench. I suggest using this to find the index. (Note: should be the same index as in `get_available_devices()`). Also we only need this here so I think this minimal effort is good enough for now.